### PR TITLE
Adding Ferrix Uprising event card to SEC set

### PR DESCRIPTION
**Pull request by mnechromancer**
## Overview

Implements the **Ferrix Uprising** event card for the SEC set as requested. This card deals damage to a unit equal to twice the number of units you control in its arena.

## Card Details

- **Set**: SEC (06_SEC)
- **Type**: Event
- **Ability**: "Deal damage to a unit equal to twice the number of units you control in its arena"

## Implementation

The card follows existing patterns from the codebase:

- **Damage multiplier pattern** (from Yoda, My Ally is the Force): Applies a 2× multiplier to the unit count
- **Arena-specific counting** (from Synchronized Strike): Counts only units in the target's specific arena (ground or space)
- **Event card structure** (from SEC event cards): Proper TypeScript implementation following the EventCard base class

### Key Features

- Targets any unit (friendly or enemy)
- Dynamically calculates damage based on the number of friendly units in the target's arena
- Works with both ground and space arenas
- Handles edge cases (zero units, targeting friendly units, controlled units)

## Files Added

1. `server/game/cards/06_SEC/events/FerrixUprising.ts` - Card implementation
2. `test/server/cards/06_SEC/events/FerrixUprising.spec.ts` - Comprehensive test suite with 6 test cases

## Testing

The test suite covers:
- Basic functionality with multiple units in ground arena
- Space arena targeting
- Edge case with zero units in target's arena
- Single unit scenarios
- Friendly unit targeting
- Unit counting after control effects (Traitorous)

## Quality Checks

- ✅ TypeScript compilation successful
- ✅ ESLint validation passed
- ✅ CodeQL security scan passed
- ✅ Card properly registered in the Index system

## Example Usage

```typescript
// Player has 2 units in ground arena, 1 in space arena
// Targets enemy ground unit
// Result: 4 damage (2 units × 2)

// Player has 3 units in space arena, 2 in ground arena  
// Targets enemy space unit
// Result: 6 damage (3 units × 2)
```

### CODE REVIEW

File: server/game/cards/06_SEC/events/FerrixUprising.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import type { IAbilityHelper } from '../../../AbilityHelper';
+import type { AbilityContext } from '../../../core/ability/AbilityContext';
+import { EventCard } from '../../../core/card/EventCard';
+import type { IEventAbilityRegistrar } from '../../../core/card/AbilityRegistrationInterfaces';
+import { WildcardCardType } from '../../../core/Constants';
+
+export default class FerrixUprising extends EventCard {
+    protected override getImplementationId() {
+        return {
+            id: 'ferrix-uprising-id',
```
[the remark this answers is not in the document]
Comment (mnechromancer) [reply server/game/cards/06_SEC/events/FerrixUprising.ts]: Thank youuuu, had that in my first iteration but totally forgot it in this one

**Review by AMMayberry1 — DISMISSED** [review]
